### PR TITLE
divided JsonHelper into translating and nontranslating versions to st…

### PR DESCRIPTION
**Pull request by RobertLaverick**
…ore untranslatedMessageJson in db

### CODE REVIEW

File: it/api/PDFRetrievalControllerISpec.scala [hunk 1]
```diff
@@ -43,8 +45,8 @@ class PDFRetrievalControllerISpec extends ApiSpecBase {
         LocalDateTime.now(),
         3,
         NonEmptyList(
-          MessageWithStatus(LocalDateTime.now(), MessageType.DepartureDeclaration, <departure></departure>, MessageStatus.SubmissionSucceeded, 1),
-          List(MessageWithoutStatus(LocalDateTime.now(), MessageType.ReleaseForTransit, <released></released>, 2))
+          MessageWithStatus(LocalDateTime.now(), MessageType.DepartureDeclaration, <departure></departure>, MessageStatus.SubmissionSucceeded, 1, convertXmlToJson(<departure></departure>.toString)),
+          List(MessageWithoutStatus(LocalDateTime.now(), MessageType.ReleaseForTransit, <released></released>, 2, convertXmlToJson(<released></released>.toString)))
```
Comment (amaalali) [inline it/api/PDFRetrievalControllerISpec.scala:49]: This can be dummy json here

File: it/api/PDFRetrievalControllerISpec.scala [hunk 2]
```diff
@@ -43,8 +45,8 @@ class PDFRetrievalControllerISpec extends ApiSpecBase {
         LocalDateTime.now(),
         3,
         NonEmptyList(
-          MessageWithStatus(LocalDateTime.now(), MessageType.DepartureDeclaration, <departure></departure>, MessageStatus.SubmissionSucceeded, 1),
-          List(MessageWithoutStatus(LocalDateTime.now(), MessageType.ReleaseForTransit, <released></released>, 2))
+          MessageWithStatus(LocalDateTime.now(), MessageType.DepartureDeclaration, <departure></departure>, MessageStatus.SubmissionSucceeded, 1, convertXmlToJson(<departure></departure>.toString)),
```
Comment (amaalali) [inline it/api/PDFRetrievalControllerISpec.scala:48]: We don't need to translate this

**Review by amaalali — CHANGES_REQUESTED** [review]
We should stick to the same architecture as in Arrivals, see [here](https://github.com/hmrc/transit-movements-trader-at-destination/blob/master/app/models/MovementMessage.scala#L40-L48).

We don't want or need to call `JsonHelper` anywhere  other than in `app/models/Message.scala`

File: app/models/Message.scala [hunk 3]
```diff
@@ -49,20 +58,39 @@ object Message extends NodeSeqFormat with MongoDateTimeFormats {
...
-  implicit val formatsMessage: OFormat[MessageWithStatus] =
-    Json.format[MessageWithStatus]
+  implicit val readsMessagWithStatus: Reads[MessageWithStatus] =
+    (
+      (__ \ "dateTime").read[LocalDateTime] and
+        (__ \ "messageType").read[MessageType] and
+        (__ \ "message").read[NodeSeq] and
+        (__ \ "status").read[MessageStatus] and
+        (__ \ "messageCorrelationId").read[Int] and
+        (__ \ "messageJson").read[JsObject].orElse(Reads.pure(Json.obj()))
```
[the remark this answers is not in the document]
Comment (RobertLaverick) [reply app/models/Message.scala:76]: Last time I checked departures backend IS live and used by the front-end. You've confused me here.

[the remark this answers is not in the document]
Comment (amaalali) [reply app/models/Message.scala:76]: Departures FE is not live but as discussed, this is for pre-prod environments where there is live data that we want to gracefully handle

File: app/models/Message.scala [hunk 4]
```diff
@@ -49,20 +58,39 @@ object Message extends NodeSeqFormat with MongoDateTimeFormats {
...
 
-  implicit val formatsMessage: OFormat[MessageWithoutStatus] =
-    Json.format[MessageWithoutStatus]
+  implicit val readsMessagWithoutStatus: Reads[MessageWithoutStatus] =
+    (
+      (__ \ "dateTime").read[LocalDateTime] and
+        (__ \ "messageType").read[MessageType] and
+        (__ \ "message").read[NodeSeq] and
+        (__ \ "messageCorrelationId").read[Int] and
+        (__ \ "messageJson").read[JsObject].orElse(Reads.pure(Json.obj()))
```
[the remark this answers is not in the document]
Comment (amaalali) [reply app/models/Message.scala:91]: Same as above

**Review by amaalali — APPROVED** [review]
I think that we can improve the code by encapsulating all XML to Json conversion in `MovementMessage` similar to what is done in Arrivals (see https://github.com/hmrc/transit-movements-trader-at-destination/blob/master/app/audit/AuditService.scala) but this is functionally correct for the current purposes. We should ideally deal with this technical debt now if possible but it can be followed up in another PR

**Review by amaalali — APPROVED** [review]

**Review by crystalmark — APPROVED** [review]

**Pull request merged**